modules/custom/activity_creator/src/ActivityNotifications.php

```diff
@@ -100,38 +156,55 @@ public function markEntityNotificationsAsRead(AccountInterface $account, EntityB
    * @param \Drupal\Core\Entity\EntityBase $entity
    *   Entity object.
    *
-   * @throws \Drupal\Core\Entity\EntityStorageException
+   * @deprecated in opensocial:8.x-7.1 and is removed from opensocial:8.x-8.0. Use
+   *   \Drupal\activity_creator\ActivityNotifications
+   * ::markEntityNotificationsAsSeen() instead.
+   *
+   * TODO: Change @see to point to a change record.
+   * @see https://www.drupal.org/project/social/issues/3087083
    */
   public function markEntityAsRead(AccountInterface $account, EntityBase $entity) {
-
     // Retrieve all the activities referring this entity for this account.
     $ids = $this->getNotificationIds($account, [ACTIVITY_STATUS_RECEIVED, ACTIVITY_STATUS_SEEN], $entity);
-
-    $activities = Activity::loadMultiple($ids);
-    foreach ($activities as $activity) {
-      $this->changeStatusOfActivity($activity, ACTIVITY_STATUS_READ);
-    }
+    $this->changeStatusOfActivity($ids, $account, ACTIVITY_STATUS_READ);
```

Comment:
I thought we removed the `ACTIVITY_STATUS_READ`? 
If I look at the code in `modules/custom/activity_creator/activity_creator.module` that is.

Also we might have to look in to changing any current activities that have the status `ACTIVITY_STATUS_READ` if we fully remove this. Otherwise we might end up with data in the database that get lost in our flow.
Imagine having to write an update hook in the future and forgetting about the fact thare are still activities with this status.

Reply:
Oh yes @ronaldtebrake I will remove it. For the latter, we are migrating all the notifications in a new table here: https://github.com/goalgorilla/open_social/pull/1547/commits/cc493a55517cf488775044cef697f745a752f761#diff-88b7e01399cd0601d630de05185566e3 , do you think this will solve the problem ?

Reply:
I don't think so, I think we will still have rogue data that uses this `ACTIVITY_STATUS_READ` value. I'd rather see this being updated to the other status.

Reply:
@ronaldtebrake This should solve it now: https://github.com/goalgorilla/open_social/pull/1547/commits/cb2ecb27a085da7c3eb6a54be6e591b99dd18f01